### PR TITLE
Added support for installation and uninstallation of aperture in aperturectl

### PR DESCRIPTION
**Pull request by hdkshingala**
### Description of change

Added support for installation for Agent and Controller via the Aperturectl so that user don't need to install helm.
Example:
```
aperturectl install controller --values-file=controller-values.yaml
aperturectl install agent --values-file=agent-values.yaml
aperturectl install istioconfig --namespace istio-system

aperturectl uninstall controller
aperturectl uninstall agent
aperturectl uninstall istioconfig
```

##### Checklist

- [x] Tested in playground or other setup
- [x] Documentation is changed or added

- - -
This change is [Reviewable](https://reviewable.io/reviews/fluxninja/aperture/1318)

### CODE REVIEW

**Review by kwapik — APPROVED** [review]
> Reviewed 32 of 32 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/1318)* status: all files reviewed, 1 unresolved discussion (waiting on @hdkshingala)

___
*[`cmd/aperturectl/cmd/installation/controller.go` line 75 at r1](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNgsDL_3_ChKs3a2uaS:-NNgsDL_3_ChKs3a2uaT:bohson) ([raw file](https://github.com/fluxninja/aperture/blob/ed751e530c1bbcec35472aec655a489022499b33/cmd/aperturectl/cmd/installation/controller.go#L75)):*
> ```Go
> 			}
> 
> 			ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
> ```

NIT: we could have the timeout as cmd line parameter, but this is not urgent.

**Review by hdkshingala — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/1318)* status: 25 of 32 files reviewed, 1 unresolved discussion (waiting on @kwapik)

___
*[`cmd/aperturectl/cmd/installation/controller.go` line 75 at r1](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNgsDL_3_ChKs3a2uaS:-NNgvt7KC-S0A3lNjLn9:bo2m9cx) ([raw file](https://github.com/fluxninja/aperture/blob/ed751e530c1bbcec35472aec655a489022499b33/cmd/aperturectl/cmd/installation/controller.go#L75)):*
<details><summary><i>Previously, kwapik (Krzysztof Kwapisiewicz) wrote…</i></summary><blockquote>

NIT: we could have the timeout as cmd line parameter, but this is not urgent.
</blockquote></details>

Made it configurable via cmd line parameter.

File: cmd/aperturectl/cmd/blueprints/list.go [hunk 1]
```diff
@@ -21,11 +22,11 @@ var listCmd = &cobra.Command{
 	Short: "List Aperture Blueprints",
 	Long: `
 Use this command to list the Aperture Blueprints which are already pulled and available in local system.`,
-	Example: `aperturectl blueprints list
+	Example: fmt.Sprintf(`aperturectl blueprints list
```
Comment (hasit) [inline cmd/aperturectl/cmd/blueprints/list.go:25]: I don't think these formatted strings are needed. These are just examples and can be any version.

**Review by hasit — CHANGES_REQUESTED** [review]

File: cmd/aperturectl/cmd/installation/install.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package installation
```
Comment (hasit) [inline cmd/aperturectl/cmd/installation/install.go:1]: How about we call this package `install`?

**Review by hasit — CHANGES_REQUESTED** [review]

File: cmd/aperturectl/cmd/blueprints/list.go (hunk 1 above)
Comment (hdkshingala) [reply cmd/aperturectl/cmd/blueprints/list.go:25]: My idea to have that is when user run --help, it shows the same version as aperturectl. We might end up where the aperturectl is moved a lot ahead in terms of version and the example keeps showing the older versions in example and someone using the example directly copy-pasting from docs.

File: cmd/aperturectl/cmd/installation/install.go (hunk 2 above)
Comment (hdkshingala) [reply cmd/aperturectl/cmd/installation/install.go:1]: It had uninstall as well so I renamed it `installation`. I was also confused between `install` and `installation`. WDYT?

**Review by kwapik — APPROVED** [review]
> Reviewed 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/1318)* status: all files reviewed, 2 unresolved discussions (waiting on @hasit)

File: cmd/aperturectl/cmd/blueprints/list.go (hunk 1 above)
Comment (hasit) [reply cmd/aperturectl/cmd/blueprints/list.go:25]: The version is blueprints from remote repo. There might be a point where we separate out `aperturectl` as a repo of its own.

File: cmd/aperturectl/cmd/installation/install.go (hunk 2 above)
Comment (hasit) [reply cmd/aperturectl/cmd/installation/install.go:1]: Sounds good.

File: cmd/aperturectl/cmd/blueprints/list.go (hunk 1 above)
Comment (hasit) [reply cmd/aperturectl/cmd/blueprints/list.go:25]: So passing `aperturectl`'s version here is pointless.

Comment (hdkshingala) [reply cmd/aperturectl/cmd/blueprints/list.go:25]: Should we remove the version from example? I don't feel showing different versions for example on docs for docs and example won't look good? WDYT?

**Review by hasit — APPROVED** [review]

File: cmd/aperturectl/cmd/blueprints/list.go (hunk 1 above)
Comment (hdkshingala) [reply cmd/aperturectl/cmd/blueprints/list.go:25]: Let's keep it there till it is under aperture repo.. will use some other way when it is taken out of it

Comment (hasit) [reply cmd/aperturectl/cmd/blueprints/list.go:25]: It is good to show what all the command is capable of.

Comment (hasit) [reply cmd/aperturectl/cmd/blueprints/list.go:25]: It clearly being an example, it should not matter what version is shown. It is under "Examples" section.

**Review by kklimonda-fn — CHANGES_REQUESTED** [review]
> Reviewed 9 of 32 files at r1, 2 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/1318)* status: all files reviewed, 11 unresolved discussions (waiting on @hdkshingala)

___
*[`docs/content/reference/aperturectl/blueprints/generate/generate.md` line 26 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNknKxx7mSIqvMnMmxQ:-NNknKxx7mSIqvMnMmxR:ba5cd0d) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/blueprints/generate/generate.md#L26)):*
> ```Markdown
> aperturectl blueprints generate --name=policies/static-rate-limiting --values-file=rate-limiting.yaml
> 
> aperturectl blueprints generate --name=policies/static-rate-limiting --values-file=rate-limiting.yaml --version v0.0.1
> ```

Nit: would it perhaps make more sense to you `latest` here (and in few other places below) as an example?
___
*[`docs/content/reference/aperturectl/install/install.md` line 24 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNknbQiASM-x--gKJtV:-NNknbQiASM-x--gKJtW:b-5xldj0) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/install.md#L24)):*
> ```Markdown
>       --namespace string     Namespace in which the component will be installed. Defaults to component name
>       --values-file string   Values YAML file containing parameters to customize the installation
>       --version string       Version of the Aperture to uninstall. Defaults to latest (default "latest")
> ```

This should probably be "install" instead?



_Code quote:_
```Markdown
uninstall
```
___
*[`docs/content/reference/aperturectl/install/install.md` line 24 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNko30s04-WNECeir0R:-NNko30s04-WNECeir0S:b-f0tigk) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/install.md#L24)):*
> ```Markdown
>       --namespace string     Namespace in which the component will be installed. Defaults to component name
>       --values-file string   Values YAML file containing parameters to customize the installation
>       --version string       Version of the Aperture to uninstall. Defaults to latest (default "latest")
> ```

I think if we use "Defaults to" everywhere else, we shouldn't be also adding (default "latest")



_Code quote:_
```Markdown
(default "latest")
```
___
*[`docs/content/reference/aperturectl/install/agent/agent.md` line 42 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNkno7R5Hnz_b-VTYYF:-NNkno7R5Hnz_b-VTYYG:buo9nmt) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/agent/agent.md#L42)):*
> ```Markdown
>       --namespace string     Namespace in which the component will be installed. Defaults to component name
>       --values-file string   Values YAML file containing parameters to customize the installation
>       --version string       Version of the Aperture to uninstall. Defaults to latest (default "latest")
> ```

Same here



_Code quote:_
```Markdown
uninstall
```
___
*[`docs/content/reference/aperturectl/install/agent/agent.md` line 42 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNknsRF5Wq6ETn3f9_r:-NNknsRF5Wq6ETn3f9_s:bb73xfv) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/agent/agent.md#L42)):*
> ```Markdown
>       --namespace string     Namespace in which the component will be installed. Defaults to component name
>       --values-file string   Values YAML file containing parameters to customize the installation
>       --version string       Version of the Aperture to uninstall. Defaults to latest (default "latest")
> ```

Same



_Code quote:_
```Markdown
(default "latest")
```
___
*[`docs/content/reference/aperturectl/install/controller/controller.md` line 42 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNko88U2WqXuGIzgG4w:-NNko88U2WqXuGIzgG4x:bhocdqu) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/controller/controller.md#L42)):*
> ```Markdown
>       --namespace string     Namespace in which the component will be installed. Defaults to component name
>       --values-file string   Values YAML file containing parameters to customize the installation
>       --version string       Version of the Aperture to uninstall. Defaults to latest (default "latest")
> ```

Same comment as for agent



_Code quote:_
```Markdown
Version of the Aperture to uninstall. Defaults to latest (default "latest")
```
___
*[`docs/content/reference/aperturectl/uninstall/uninstall.md` line 24 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNkoCQnDbhhFBWdGBIk:-NNkoCQnDbhhFBWdGBIl:bg7eat0) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/uninstall/uninstall.md#L24)):*
> ```Markdown
>       --namespace string     Namespace from which the component will be uninstalled. Defaults to component name
>       --timeout int          Timeout of waiting for uninstallation hooks completion. Defaults to 300 (default 300)
>       --version string       Version of the Aperture to uninstall. Defaults to latest (default "latest")
> ```

Same comment as for installation - I think it should be either either "Defaults to" or "(default "latest")" but not both



_Code quote:_
```Markdown
(default "latest")
```
___
*[`docs/content/reference/aperturectl/uninstall/agent/agent.md` line 41 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNkoLqf2F3cI8rFvgYs:-NNkoLqf2F3cI8rFvgYt:buo9nmt) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/uninstall/agent/agent.md#L41)):*
> ```Markdown
>       --namespace string     Namespace from which the component will be uninstalled. Defaults to component name
>       --timeout int          Timeout of waiting for uninstallation hooks completion. Defaults to 300 (default 300)
>       --version string       Version of the Aperture to uninstall. Defaults to latest (default "latest")
> ```

Same here



_Code quote:_
```Markdown
(default "latest")
```
___
*[`docs/content/reference/aperturectl/uninstall/controller/controller.md` line 41 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNkoOY645WwHiYi--36:-NNkoOY645WwHiYi--37:buo9nmt) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/uninstall/controller/controller.md#L41)):*
> ```Markdown
>       --namespace string     Namespace from which the component will be uninstalled. Defaults to component name
>       --timeout int          Timeout of waiting for uninstallation hooks completion. Defaults to 300 (default 300)
>       --version string       Version of the Aperture to uninstall. Defaults to latest (default "latest")
> ```

Same here



_Code quote:_
```Markdown
(default "latest")
```

**Review by hdkshingala — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/1318)* status: all files reviewed, 11 unresolved discussions (waiting on @kklimonda-fn)

___
*[`docs/content/reference/aperturectl/blueprints/generate/generate.md` line 26 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNknKxx7mSIqvMnMmxQ:-NNkuHQ3B0GZR2gtdTsS:bfaf43c) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/blueprints/generate/generate.md#L26)):*
<details><summary><i>Previously, kklimonda-fn (Krzysztof Klimonda) wrote…</i></summary><blockquote>

Nit: would it perhaps make more sense to you `latest` here (and in few other places below) as an example?
</blockquote></details>

Using `latest` everywhere in examples.
___
*[`docs/content/reference/aperturectl/install/install.md` line 24 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNknbQiASM-x--gKJtV:-NNkuLwLDvUYdgzOYJTu:b-50bfj4) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/install.md#L24)):*
<details><summary><i>Previously, kklimonda-fn (Krzysztof Klimonda) wrote…</i></summary><blockquote>

This should probably be "install" instead?
</blockquote></details>

Will re-phrase without install/uninstall
___
*[`docs/content/reference/aperturectl/install/install.md` line 24 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNko30s04-WNECeir0R:-NNkuQRrECwUJI3tJCXZ:b-1c5mtj) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/install.md#L24)):*
<details><summary><i>Previously, kklimonda-fn (Krzysztof Klimonda) wrote…</i></summary><blockquote>

I think if we use "Defaults to" everywhere else, we shouldn't be also adding (default "latest")
</blockquote></details>

Yes this came to my notice also when I was working on docs. Will be fixed in next commit.
___
*[`docs/content/reference/aperturectl/install/agent/agent.md` line 42 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNknsRF5Wq6ETn3f9_r:-NNkubW_9UXOCGcu0Zbr:b-1c5mtj) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/agent/agent.md#L42)):*
<details><summary><i>Previously, kklimonda-fn (Krzysztof Klimonda) wrote…</i></summary><blockquote>

Same
</blockquote></details>

Yes this came to my notice also when I was working on docs. Will be fixed in next commit.
___
*[`docs/content/reference/aperturectl/install/agent/agent.md` line 42 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNkno7R5Hnz_b-VTYYF:-NNkugh95gej6matNFC6:b-50bfj4) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/agent/agent.md#L42)):*
<details><summary><i>Previously, kklimonda-fn (Krzysztof Klimonda) wrote…</i></summary><blockquote>

Same here
</blockquote></details>

Will re-phrase without install/uninstall
___
*[`docs/content/reference/aperturectl/install/controller/controller.md` line 42 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNko88U2WqXuGIzgG4w:-NNkuiHa4odsddjqYmTL:b-50bfj4) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/install/controller/controller.md#L42)):*
<details><summary><i>Previously, kklimonda-fn (Krzysztof Klimonda) wrote…</i></summary><blockquote>

Same comment as for agent
</blockquote></details>

Will re-phrase without install/uninstall
___
*[`docs/content/reference/aperturectl/uninstall/uninstall.md` line 24 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNkoCQnDbhhFBWdGBIk:-NNkucleCXJRxi8hAAz-:b-1c5mtj) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/uninstall/uninstall.md#L24)):*
<details><summary><i>Previously, kklimonda-fn (Krzysztof Klimonda) wrote…</i></summary><blockquote>

Same comment as for installation - I think it should be either either "Defaults to" or "(default "latest")" but not both
</blockquote></details>

Yes this came to my notice also when I was working on docs. Will be fixed in next commit.
___
*[`docs/content/reference/aperturectl/uninstall/agent/agent.md` line 41 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNkoLqf2F3cI8rFvgYs:-NNkudKv1JXVwjPPXtF9:b-1c5mtj) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/uninstall/agent/agent.md#L41)):*
<details><summary><i>Previously, kklimonda-fn (Krzysztof Klimonda) wrote…</i></summary><blockquote>

Same here
</blockquote></details>

Yes this came to my notice also when I was working on docs. Will be fixed in next commit.
___
*[`docs/content/reference/aperturectl/uninstall/controller/controller.md` line 41 at r2](https://reviewable.io/reviews/fluxninja/aperture/1318#-NNkoOY645WwHiYi--36:-NNkudv8BPTtdrspTBhP:b-1c5mtj) ([raw file](https://github.com/fluxninja/aperture/blob/8be2f8d461a32014121e2c453521cc05dd6bd10c/docs/content/reference/aperturectl/uninstall/controller/controller.md#L41)):*
<details><summary><i>Previously, kklimonda-fn (Krzysztof Klimonda) wrote…</i></summary><blockquote>

Same here
</blockquote></details>

Yes this came to my notice also when I was working on docs. Will be fixed in next commit.

**Review by kklimonda-fn — APPROVED** [review]
> Reviewed 13 of 32 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/fluxninja/aperture/1318)* status: 23 of 42 files reviewed, 2 unresolved discussions (waiting on @hasit, @hdkshingala, and @kwapik)

**Pull request merged**